File: src/main/java/org/slimefunguguproject/bump/implementation/items/RandomEquipment.java
package org.slimefunguguproject.bump.implementation.items;

import javax.annotation.ParametersAreNonnullByDefault;

import org.bukkit.inventory.ItemStack;

import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup;
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem;
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack;
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;

import org.slimefunguguproject.bump.core.attributes.AppraisableItem;

/**
 * A {@link RandomEquipment} is an item that is an {@link AppraisableItem}.
 *
 * @author ybw0014
 */
public final class RandomEquipment extends SlimefunItem implements AppraisableItem {

    @ParametersAreNonnullByDefault
    public RandomEquipment(ItemGroup itemGroup, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
        super(itemGroup, item, recipeType, recipe);
    }
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/implementation/items/RandomEquipment.kt
package io.github.slimefunguguproject.bump.implementation.items

import io.github.slimefunguguproject.bump.core.attributes.AppraisableItem
import io.github.slimefunguguproject.bump.implementation.BumpItems
import io.github.thebusybiscuit.slimefun4.api.items.ItemGroup
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItem
import io.github.thebusybiscuit.slimefun4.api.items.SlimefunItemStack
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun4.implementation.items.multiblocks.Compressor
import org.bukkit.inventory.ItemStack

class RandomEquipment(
    itemGroup: ItemGroup,
    itemStack: SlimefunItemStack,
    recipeType: RecipeType,
    recipe: Array<out ItemStack?>,
) : SlimefunItem(itemGroup, itemStack, recipeType, recipe), AppraisableItem {
    override fun postRegister() {
        val compressor: Compressor = (getById("COMPRESSOR") ?: return) as Compressor
        compressor.addRecipe(arrayOf(item), BumpItems.COMPRESSED_RANDOM_EQUIPMENT)
    }
}
